pkg/iac/terraform/resource_block.go

```diff
@@ -108,6 +103,24 @@ EOF
 
 }
 
+func parseStringPrimitive(input string) string {
+	// we must escape templating
+	// ref: https://developer.hashicorp.com/terraform/language/expressions/strings#escape-sequences-1
+	r := regexp.MustCompile(`((\$|\%)\{.+\})`)
+	ff := r.ReplaceAllStringFunc(input, func(s string) string {
+		s = strings.ReplaceAll(s, "$", "$$")
+		s = strings.ReplaceAll(s, "%", "%%")
```

Comment:
Only the first `$` or `%` character should be escaped, as this character may occur in the string. For example: `${test$} -> $${test$}`